### PR TITLE
[Merged by Bors] - feat: generalize `Prod.wellFoundedLT/GT` to `Preorder`s

### PR DESCRIPTION
**Pull request by alreadydone**
.. using a standard antisymmetrization trick. Also

+ Add lemmas `wellFoundedLT/GT_antisymmetrization_iff`.

+ Establish the `OrderIso` between the antisymmetrization of a product preorder and the product of antisymmetrizations.

+ Golf the original `Prod.wellFoundedLT` and rename it to `Prod.wellFoundedLT'`.

+ Golf `StrictMono.wellFoundedLT` etc. and move to where `StrictMono` is defined.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Antisymmetrization.lean [hunk 1]
```diff
@@ -233,3 +247,22 @@ theorem OrderIso.dualAntisymmetrization_symm_apply (a : α) :
   rfl
 
 end Preorder
+
+section Prod
+
+variable (α β) [Preorder α] [Preorder β]
+
+/-- The antisymmetrization of a product preorder is isomorphic
```
Comment (vihdzp) [inline Mathlib/Order/Antisymmetrization.lean:255]:
```suggestion
/-- The antisymmetrization of a product preorder is order isomorphic
```

File: Mathlib/Order/Antisymmetrization.lean [hunk 2]
```diff
@@ -233,3 +247,22 @@ theorem OrderIso.dualAntisymmetrization_symm_apply (a : α) :
...
+to the product of antisymmetrizations. -/
+def Antisymmetrization.prodEquiv :
```
Comment (vihdzp) [inline Mathlib/Order/Antisymmetrization.lean:257]: Does it make sense to provide `simp` lemmas here? e.g. `Antisymmetrization.prodEquiv ⟦x⟧ = (⟦x.1⟧, ⟦x.2⟧)`, etc.

File: Mathlib/Order/Cover.lean [hunk 3]
```diff
@@ -550,6 +550,22 @@ theorem covBy_iff : x ⋖ y ↔ x.1 ⋖ y.1 ∧ x.2 = y.2 ∨ x.2 ⋖ y.2 ∧ x.
 
 end Prod
 
+namespace Prod
```
Comment (vihdzp) [inline Mathlib/Order/Cover.lean:553]: This feels like a somewhat random file to put these instances in. What's the reasoning?

**Review by vihdzp — COMMENTED** [review]
Nice!

File: Mathlib/Order/Cover.lean (hunk 3 above)
Comment (alreadydone) [reply Mathlib/Order/Cover.lean:553]: I'm using both StrictMono.wellFoundedLT into Antisymmetrization in the proofs, which don't import each other. When I tried to imported StrictMono.wellFoundedLT into Antisymmetrization I got +16.91% increase in transitive imports and was [warned by the bot](https://github.com/leanprover-community/mathlib4/pull/17435#issuecomment-2394988628). `#find_home` found this order theory file that imports both, but now I've figured out the correct solution is to golf the proof of StrictMono.wellFoundedLT and move it to where StrictMono is defined ...

File: Mathlib/Order/Antisymmetrization.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/Order/Antisymmetrization.lean:255]: done

File: Mathlib/Order/Antisymmetrization.lean (hunk 2 above)
Comment (alreadydone) [reply Mathlib/Order/Antisymmetrization.lean:257]: added two lemmas

**Review by alreadydone — COMMENTED** [review]
Thanks for the review!

**Review by vihdzp — APPROVED** [review]
LGTM!

**Review by YaelDillies — APPROVED** [review]
Nicely spotted! It's kind of surprising having this instance in Order.antisymmetrzaion  ut I huess that's fine for now.

maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**